provider.go

```diff
@@ -54,12 +88,30 @@ func providerConfig(d *schema.ResourceData) (interface{}, error) {
 
 	var sess *session.Session
 	var err error
-	if profile != defaultAWSProfile {
+	var creds *credentials.Credentials
+
+	assumeRoleList := d.Get("assume_role").(*schema.Set).List()
+
+    if len(assumeRoleList) == 1 {
+
+        assumeRole := assumeRoleList[0].(map[string]interface{})
+        sess, err = session.NewSession(&aws.Config{Region: aws.String(region)})
+
+        creds = stscreds.NewCredentials(sess, assumeRole["role_arn"].(string), func(arp *stscreds.AssumeRoleProvider) {
+            arp.RoleSessionName = assumeRole["session_name"].(string)
+            arp.Duration = 60 * time.Minute
+            arp.ExpiryWindow = 30 * time.Second
+          })
+        return credstash.NewWithAssumedRole(table, sess, creds), nil
```

Comment:
By returning early here the `profile` is being ignored. Also the error from line 98 is being discarded. Please merge the different branches where the AWS session is created. I think a single `session.Options` struct could be created and set up. It would be nice if different config options would not interfere or disable each other by chance.